src/components/visit/Modal.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react'
+import { useState } from 'react'
+
+import { User, writeUserData } from '@/../databaseIntigration'
+import { CancelButton } from '@/components/visit/Buttons'
+import FormField from '@/components/visit/formField'
+
+interface ModalViewProps {
+  openFunc: () => void
+}
+
+const ModalView: React.FC<ModalViewProps> = ({ openFunc }) => {
+  const [firstName, setFirstName] = useState('')
+  const [lastName, setLastName] = useState('')
+  const [petName, setPetName] = useState('')
+  const [dateTime, setDateTime] = useState('')
+  const [duration, setDuration] = useState('')
+  const [walkDist, setWalkDist] = useState(0)
+  const [commuteDist, setCommuteDist] = useState(0)
+  const [commuteMethod, setCommuteMethod] = useState('')
+  const [notes, setNotes] = useState('')
+
+  return (
+    <div className='fixed inset-0 z-50 h-screen w-screen rounded-sm bg-white p-4 shadow '>
+      <div>
+        <div className='fixed right-2 top-2 h-7 w-7 rounded-full bg-primary'>
+          <button onClick={openFunc}>
+            <CancelButton />
+          </button>
+        </div>
+
+        <h1 className='mx-1 border-b-2 border-primary py-3 pt-10 text-2xl font-bold'>
+          Add Your Visit
+        </h1>
+        <form
+          className='pt-3'
+          onSubmit={() => {
+            const data: User = {
+              firstName: firstName,
+              lastName: lastName,
+              petName: petName,
+              dateTime: dateTime,
+              duration: duration,
+              walkDist: walkDist,
+              commuteDist: commuteDist,
+              commuteMethod: commuteMethod,
+              notes: notes
+            }
+            writeUserData(data)
```

Comment:
Generally it's cleaner to try split up your presentation code and the logic that powers it. I would extract this onSubmit function out, maybe something like this:

```tsx
const onSubmit = () => {
    const data: User = {
        ...
    }
    writeUserData(data);
};

return (
    ...
    <form onSubmit={onSubmit} ... />
    ...
);
```